client/test/unit/components/bhLocationSelect.spec.js

```diff
@@ -0,0 +1,201 @@
+
+  beforeEach(inject(function (_$rootScope_, _$compile_, _$httpBackend_) {
+    scope = _$rootScope_.$new();
+    $compile = _$compile_;
+    $httpBackend = _$httpBackend_;
+
+    $httpBackend.whenGET('/locations/countries').respond(testData.countries);
+    $httpBackend.whenRoute('GET', '/locations/detail/:uuid')
+      .respond(function (method, url, data, headers, params) {
+        var response = {};
+
+        var village =  testData.villages.filter(function (village){ return village.uuid === params.uuid; })[0];
+        var sector = testData.sectors.filter(function (sector){ return sector.uuid === village.sector_uuid; })[0];
+        var province = testData.provinces.filter(function (province){ return province.uuid === sector.province_uuid; })[0];
+        var country = testData.countries.filter(function (country){ return country.uuid === province.country_uuid; })[0];
+
+        response.villageUuid = village.uuid;
+        response.village = village.name;
+        response.sector = sector.name;
+        response.sectorUuid = sector.uuid;
+        response.province = province.name;
+        response.provinceUuid = province.uuid;
+        response.country = country.name;
+        response.countryUuid = country.uuid;
+
+        return [200, response];
+      });
+    $httpBackend.whenRoute('GET', '/locations/provinces')
+      .respond(function (method, url, data, headers, params) {
+        var response;
+        if(params.country){
+          response = testData.provinces.filter(function (province){ return province.country_uuid === params.country;});
+        }else{
+          response = testData.provinces;
+        }
+
+        return [200, response];
+      });
+    $httpBackend.whenRoute('GET', '/locations/sectors')
+      .respond(function (method, url, data, headers, params) {
+        var response;
+        if(params.province){
+          response = testData.sectors.filter(function (sector){ return sector.province_uuid === params.province;});
+        }else{
+          response = testData.sectors;
+        }
+
+        return [200, response];
+      });
+    $httpBackend.whenRoute('GET', '/locations/villages')
+      .respond(function (method, url, data, headers, params) {
+        var response;
+        if(params.sector){
+          response = testData.villages.filter(function (village){ return village.sector_uuid === params.sector;});
+        }else{
+          response = testData.villages;
+        }
+
+        return [200, response];
+      });
+  }));
+
+  afterEach(function() {
+    $httpBackend.verifyNoOutstandingExpectation();
+    $httpBackend.verifyNoOutstandingRequest();
+  });
+
+  it('Loads an initial village, sector, province, country, countries and country message correctly', function (){
+    //This test case test indirectly the loadLocation and the loadCountries functions
+    scope.testData = testData;
+    element = $compile(angular.element(template))(scope);
```

Comment:
Every test does:
```js
scope.testData = testData;
element = $compile(angular.element(template))(scope);
scope.$digest();
componentController = element.controller('bhLocationSelect');
```
This looks like setup code.  Could you put it in `beforeEach()` block?